src/Auth/JwtAuth.php

```diff
@@ -19,53 +19,38 @@ class JwtAuth implements IAuth {
      */
     public $key;
 
-    /**
-     * @var string|null
-     */
-    public $token;
-
     /**
      * JwtAuth constructor.
      * @param array|string $claim_or_token
      * @param string|null $key
      */
-    public function __construct( $claim_or_token, string $key = null ) {
-        if (is_null($key)) {
-            $this->claim = null;
-            $this->key = null;
-            $this->token = $claim_or_token;
-        } else {
-            $this->claim = $claim_or_token;
-            $this->key = $key;
-            $this->token = null;
-        }
+    public function __construct( $claim, string $key ) {
+        $this->claim = $claim;
+        $this->key = $key;
     }
 
     function build_header(): string {
+        $claim = array_merge( $this->claim,
+            [
+                'exp' => time() + 60 * 10, // 10 minutes
+            ]
+        );
 
-        if (!is_null($this->token)) {
-            $token = $this->token;
-        } else {
-            $claim = array_merge( $this->claim,
-                [
-                    'exp' => time() + 60 * 10, // 10 minutes
-                ]
-            );
-
-            $token = JWT::encode($claim, $this->key);
-        }
+        $token = JWT::encode($claim, $this->key);
 
         return "Bearer " . $token;
-
     }
 
-    static function validate_signature( $token, $key ): bool {
+    static function validate_signature( $token, $key, $iss = null): bool {
         try {
             $claim = JWT::decode( $token, $key, [ 'HS256' ] );
         } catch( Throwable $ex ) {
             return false;
         }
 
+        if( gettype( $claim ) === 'string' || $claim->iss !== $iss ) {
```

Comment:
```suggestion
        if( $iss !== NULL && ( !property_exists( $claim,'iss' ) || $claim->iss !== $iss ) ) {
```

We only want to do this check if `$iss` was passed in.

`JWT::decode` is guaranteed to return an object if it didn't throw an exception.
However, if you try to read `->iss` on an object that doesn't have that field, it can raise `PHP Notice:  Undefined property`, so we make sure it's there.